buzz_el/entity_matcher/entity_matcher.py

```diff
@@ -0,0 +1,158 @@
+from typing import Dict, Iterable, List, Optional
+
+from spacy.language import Language
+from spacy.pipeline import SpanRuler
+from spacy.tokens import Doc
+
+from ..graph_loader import GraphLoader
+
+
+class EntityMatcher:
+    """
+    A class to construct a spaCy span ruler from a RDF graph.
+
+    Parameters
+    ----------
+    graph_loader : GraphLoader
+        An instance of the GraphLoader class for loading knowledge graph data.
+    spacy_model : Language
+        The spaCy language to use when constructing the spaCy span ruler.
+    span_ruler_config : Optional[Dict], optional
+        Configuration for the spaCy span ruler, by default {"phrase_matcher_attr": "LOWER"}.
+        See <https://spacy.io/api/spanruler#config>
+
+    Attributes
+    ----------
+    spacy_model : Language
+        The spaCy language to use when constructing the spaCy span ruler.
+    graph_loader : GraphLoader
+        An instance of the GraphLoader class for loading knowledge graph data.
+    _span_ruler_config : Dict
+        Configuration for the spaCy span ruler, by default {"phrase_matcher_attr": "LOWER"}.
+        See: <https://spacy.io/api/spanruler#config>
+    spacy_component : SpanRuler
+        The spaCy span ruler component for the knowledge graph entity matching.
+    """
+
+    def __init__(
+        self,
+        graph_loader: GraphLoader,
```

Comment:
I think that the graph loader should be in the entity linker, not in the entity matcher

Reply:
To me the entity matcher en the entity linker are sperate.